attiny84_sketch/pidog2/pidog2.ino

```diff
@@ -87,7 +91,16 @@ void doSecondWork() {
     uint32_t s = rf.get(REG_STATUS);
     if ((s & _BV(STAT_WDOG_EN)) && (s & _BV(STAT_PWR_ON))) {
         reg_t on_rem = rf.get(REG_ON_REMAINING);
-        if (!on_rem) {
+        //reg_t a_off_thresh = rf.gethl(REG_VSENSE_OFF_THRESHOLD,register_top); <-- This always works.
+        //reg_t b_off_thresh = rf.gethl(REG_VSENSE_OFF_THRESHOLD,register_bottom);<-- This only works when the top 16 bits are 0. Otherwise, this call returns all 32 bits.
```

Comment:
You found a bug, thanks!

I think this is happening because the compiler is treating a number as signed so when shifting right it just keeps sign extending with more 1's.

This should fix it:
```c++
REGW gethl(uint8_t addr, register_half_t half) {
    REGW v = registers[addr % REG_COUNT];
    size_t sz = sizeof(REGW);
    if (half != register_bottom) {
        v >>= (sz * 4);
    }
    REGW halfmask = ~(0x0);
    halfmask >>= (sz * 4);
    v &= halfmask;
    return v;
}
```

Would you be willing to try it out?

Reply:
Yes, I'll give this a try.

Reply:
Verified that this change fixes the bug in gethl().